src/services/pacs_integration.tsx

```diff
@@ -0,0 +1,111 @@
+import { DcmImage, PACSMainResponse } from "../context/reducers/dicomImagesReducer";
+import axios from 'axios';
+
+declare var process: {
+    env: {
+        REACT_APP_CHRIS_UI_PFDCM_URL: string,
+    }
+};
+
+class PACSIntegration {
+
+    private static basePACSFilePath = 'SERVICES/PACS/covidnet/';
+
+    static parseResponse(rawResponse: string) {
+        let responseHeader = rawResponse.split('\r');
+
+        // the 4th element of responseHeader contains the stringified JSON body return payload
+        const responseBody = responseHeader[4];
+        const responseJSON = JSON.parse(responseBody);
+        return responseJSON;
+    }
+
+    /**
+     * Performs a PACS query on the given PatientID and returns an array of DcmImages containing metadata 
+     * of the DICOM files associated with that patient
+     * @param {string | undefined} patientID 
+     */
+    static async queryPatientFiles(patientID: string | undefined): Promise<DcmImage[]> {
+        if (!patientID) return [];
+        
+        let patientData: DcmImage[] = [];
+        try {
+            const rawResponse = await axios.post(process.env.REACT_APP_CHRIS_UI_PFDCM_URL, JSON.stringify({
+                action:"PACSinteract",
+                meta:{
+                    do:"query",
+                    on:{PatientID: patientID},
+                    PACS:"orthanc"
+                }
+            }), {headers: {'Content-Type': 'text/plain'}})
+            const parsedResponse = this.parseResponse(rawResponse.data);
+            const data: PACSMainResponse[] = parsedResponse.query.data;
+            data.forEach(study => {
+                study.series.forEach(series => {
+                    patientData.push({
```

Comment:
Instead of pushing into array, I suggest using `map()` and save the results of map into `patientData`.